framework/contracts/native/version-control/src/commands.rs

```diff
@@ -31,32 +35,95 @@ use crate::{
 };
 
 /// Add new Account to version control contract
-/// Only Factory can add Account
+/// Only Account can add itself.
 pub fn add_account(
     deps: DepsMut,
     msg_info: MessageInfo,
-    account_id: AccountId,
-    account_base: Account,
     namespace: Option<String>,
+    creator: String,
 ) -> VCResult {
     let config = CONFIG.load(deps.storage)?;
 
-    // Only Factory can add new Account
-    let is_factory = config
-        .account_factory_address
-        .map(|addr| addr == msg_info.sender)
-        .unwrap_or(false);
-    if !is_factory {
-        return Err(VCError::NotAccountFactory {});
-    }
+    // Check that sender is account
+    let contract_info = cw2::query_contract_info(&deps.querier, &msg_info.sender)?;
+    let maybe_acc_module_info = ModuleInfo::try_from(contract_info)?;
+
+    ensure!(
+        maybe_acc_module_info.id() == ACCOUNT,
+        VCError::NotAccountInfo {
+            caller_info: maybe_acc_module_info
+        }
+    );
 
-    // Check if account already exists
+    // Ensure account isn't already registered
+    let account_id = ACCOUNT_ID.query(&deps.querier, msg_info.sender.clone())?;
     ensure!(
         !ACCOUNT_ADDRESSES.has(deps.storage, &account_id),
         VCError::AccountAlreadyExists(account_id)
     );
 
-    ACCOUNT_ADDRESSES.save(deps.storage, &account_id, &account_base)?;
+    let acc_module_info = maybe_acc_module_info;
+
+    let account_code_id = REGISTERED_MODULES
+        .load(deps.storage, &acc_module_info)?
+        .unwrap_account()?;
+
+    // provided and smaller, assert is eq to sequence
+    // provided and larger, just register
```

Comment:
This logic should be removed if we can't create account without instantiate2, and have sequences for all accounts

Reply:
Will do in different PR
[ORC-169: Ensure Account migration is properly protected](https://linear.app/abstract-sdk/issue/ORC-169/ensure-account-migration-is-properly-protected)